src/js/swipe.js

```diff
@@ -0,0 +1,97 @@
+import { EVENTS } from './ui-events';
+import on from './on';
+
+class Swipe {
+  constructor(element) {
+    this.xDown = null;
+    this.yDown = null;
+    this.element = typeof (element) === 'string' ? document.querySelector(element) : element;
+    // 10% of the width is the threshold swipe
+    this.DIFF_THRESHOLD = element.offsetWidth / 10;
+    this.TIME_THRESHOLD = 200;
```

Comment:
this theshold prevents all swipes to be detected for me. I think it should be raised to at least 500ms. 
Why do we need to check for the time in first place?
We can reset the Swipe on every touchStart event

Reply:
i think to distinguish swipe vs touch

Reply:
Using just time to detect a swipe is not ideal, a half second for a swiping interaction can be too long.
Best is to track direction and distance, like a vector, and how much time it took.

Reply:
I am tracking direction, distance and time.
- The direction must be horizontal with angle 45 degrees at most.
 - The distance must be at least 10% of the slider's width
 - The time must be below 200ms.
I did the research online and these values seem to be most commonly used.
Please let me know if you want these values changed.